app/src/main/java/com/trainstation/frettirtest2/service/NewsService.java

```diff
@@ -11,6 +11,7 @@
  */
 public interface NewsService
 {
-    @GET("/search?term=beyonce&entity=musicVideo")
-    Observable<NewsResponse> getNewsData();
+    @GET("/search")
+    Observable<NewsResponse> getNewsData(@Query("term") String termSearch,
+                                         @Query("entity") String entitySearch);
```

Comment:
okay here is some refactoring, it doesn't change anything how the app works, just showing how you can use retrofit (which is the library that we use to declare an endpoint like we are doing here) and how it can be more flexible :+1: 

so before the query paramters were hardcoded int the path to the endpoint: `/search?term=beyonce&entity=musicVideo`
but we can make this more flexible, by taking the query parameters as a parameter of the method,
"So what does `@Query("term") String termSearch` mean?" it adds a query parameter called `term` and the value for that query paramter is the value of the string `termSearch`. And remember that `getNewsData ` is just a normal method, so when we call this method, we need to pass in this string, and we do it here: https://github.com/helgio/frettaApp/pull/2/files#diff-3d0b8bc372ffdc0af39ccd4f1c52d8e4R30

**so why is this flexible?** e.g. if you want to search by different query parameters for different fragments (football news, business news, ect.) and maybe you want to let the user write what he/she want to see, and then you might want to take the string that the user wrote and put that into the endpoint.

please comment if there is something that you don't fully understand, and I will gladly help to explain better:)